specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc

```diff
@@ -1504,8 +1504,19 @@ delivered out of order. When incrementing the bdSeq number, if the previous valu
 must be zero.
 * [tck-testable tck-id-payloads-ndeath-will-message-publisher]#[yellow-background]*An NDEATH message
 SHOULD be published by the Edge Node before it intentionally disconnects from the MQTT Server.*#
-This allows Host Applications advanced notice that an Edge Node has disconnected rather than waiting
-for the NDEATH to be delivered by the MQTT Server based on an MQTT keep alive timeout.
+** This allows Host Applications advanced notice that an Edge Node has disconnected rather than
+waiting for the NDEATH to be delivered by the MQTT Server based on an MQTT keep alive timeout.
+* [tck-testable tck-id-payloads-ndeath-will-message-publisher-disconnect-mqtt311]#[yellow-background]*If
+the Edge Node is using MQTT 3.1.1 and it sends an MQTT DISCONNECT packet, the Edge Node MUST publish
+an NDEATH message to the MQTT Server before it sends the MQTT DISCONNECT packet.*#
+** This is to ensure Host Applications are notified that the Edge Node is disconnecting. Because an
+MQTT DISCONNECT packet is sent, the MQTT Server will not deliver the Will Message/NDEATH on behalf
+of the disconnecting Edge Node.
+* [tck-testable tck-id-payloads-ndeath-will-message-publisher-disconnect-mqtt50]#[yellow-background]*If
+the Edge Node is using MQTT 5.0 and it sends an MQTT DISCONNECT packet, the MQTT v5.0 'Disconnect
+with Will Message' flag MUST be set in the DISCONNECT packet.*#
```

Comment:
"Disconnect with Will Message" is a reason code not a flag so probably clearer to use that term.